apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/ApolloPlugin.kt

```diff
@@ -197,7 +197,20 @@ open class ApolloPlugin : Plugin<Project> {
       it.generateKotlinModels = extension.generateKotlinModels
       it.generateVisitorForPolymorphicDatatypes = extension.generateVisitorForPolymorphicDatatypes
       it.customTypeMapping = extension.customTypeMapping
-      compilationUnit.setupOutputProperties(it, extension.generateTransformedQueries, project)
+      it.outputDir.apply {
+        set(compilationUnit.outputDirectory)
+        disallowChanges()
+      }
+      it.transformedQueriesOutputDir.apply {
+        if (extension.generateTransformedQueries) {
+          set(compilationUnit.transformedQueriesDirectory)
+        }
+        disallowChanges()
+      }
     }
+    compilationUnit.outputDir = taskProvider.flatMap { it.outputDir }
+    compilationUnit.transformedQueriesDir = taskProvider.flatMap { it.transformedQueriesOutputDir }
```

Comment:
Will Groovy accept `set()` calls by auto casting CompilationUnit to DefaultCompilationUnit? If so, may be good to call `disallowChanges()` on these.

Reply:
Never mind, that shouldn't be possible, they don't even have `set()` methods!

Reply:
Yes, Providers are read only, Properties are read-write